src/crypto-utils.ts

```diff
@@ -0,0 +1,19 @@
+import CryptoJS from 'crypto-js';
+
+const arrayBufferToHex = (arrayBuffer: ArrayBuffer): string => {
+  const uint8Array = new Uint8Array(arrayBuffer);
+  let hexString = '';
+  for (const i of uint8Array) {
+    const val = uint8Array[i]?.toString(16).padStart(2, '0');
+    hexString += val;
```

Comment:
Hmm, this seems wrong. When using `for...of`, you should be getting the values in the array, not the indices. Therefore I would think this should be something like:

``` suggestion
  for (const value of uint8Array) {
    const paddedValue = value.toString(16).padStart(2, '0');
    hexString += paddedValue;
```